src/Microsoft.Azure.WebJobs.Host/Scale/TriggerMetadata.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    /// <summary>
+    /// Represents trigger metadata.
+    /// </summary>
+    public class TriggerMetadata
+    {
+        public TriggerMetadata(JObject metadata)
+            : this(metadata, new Dictionary<string, object>())
+        {
+        }
+
+        public TriggerMetadata(JObject metadata, IDictionary<string, object> properties)
+        {
+            Metadata = metadata;
+            Properties = properties;
+        }
+
+        /// <summary>
+        /// Gets the name of the Function this trigger belongs to.
+        /// </summary>
+        public string FunctionName => Metadata.GetValue("functionName", StringComparison.OrdinalIgnoreCase)?.Value<string>();
```

Comment:
For these properties that are derived from the metadata payload, shouldn't we just initialize these values once in the ctor?